src/EditorFeatures/Core/Implementation/BraceMatching/BraceHighlightingViewTaggerProvider.cs

```diff
@@ -19,44 +22,64 @@ namespace Microsoft.CodeAnalysis.Editor.Implementation.BraceMatching
     [Export(typeof(IViewTaggerProvider))]
     [ContentType(ContentTypeNames.RoslynContentType)]
     [TagType(typeof(BraceHighlightTag))]
-    internal class BraceHighlightingViewTaggerProvider :
-        AbstractAsynchronousViewTaggerProvider<BraceHighlightTag>
+    internal class BraceHighlightingViewTaggerProvider : AsynchronousTaggerProvider<BraceHighlightTag>
     {
         private readonly IBraceMatchingService _braceMatcherService;
 
+        public override IEnumerable<Option<bool>> Options => SpecializedCollections.SingletonEnumerable(InternalFeatureOnOffOptions.BraceMatching);
+
         [ImportingConstructor]
         public BraceHighlightingViewTaggerProvider(
-            IForegroundNotificationService notificationService,
             IBraceMatchingService braceMatcherService,
+            IForegroundNotificationService notificationService,
             [ImportMany] IEnumerable<Lazy<IAsynchronousOperationListener, FeatureMetadata>> asyncListeners)
-            : base(new AggregateAsynchronousOperationListener(asyncListeners, FeatureAttribute.BraceHighlighting),
-                   notificationService)
+                : base(new AggregateAsynchronousOperationListener(asyncListeners, FeatureAttribute.BraceHighlighting), notificationService)
         {
             _braceMatcherService = braceMatcherService;
         }
 
-        protected override bool RemoveTagsThatIntersectEdits => true;
-
-        protected override SpanTrackingMode SpanTrackingMode => SpanTrackingMode.EdgeExclusive;
+        public override ITaggerEventSource CreateEventSource(ITextView textView, ITextBuffer subjectBuffer)
+        {
+            return TaggerEventSources.Compose(
+                TaggerEventSources.OnTextChanged(subjectBuffer, TaggerDelay.NearImmediate),
+                TaggerEventSources.OnCaretPositionChanged(textView, subjectBuffer, TaggerDelay.NearImmediate));
```

Comment:
Not your bug, but this also needs to watch ParseOptionChangedEventSource. I suspect we should have an OnTreeChanged event source that encompasses these two.